machine_learning/benchmarks/opflow_qnn_regressor_benchmark.py

```diff
@@ -0,0 +1,132 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2021.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+"""Opflow based Neural Network Regressor benchmarks."""
+
+import pickle
+from itertools import product
+from timeit import timeit
+
+from qiskit.algorithms.optimizers.cobyla import COBYLA
+from sklearn.metrics import mean_absolute_error, mean_squared_error
+
+from .base_regressor_benchmark import (
+    BaseRegressorBenchmark,
+    DATASET_SYNTHETIC_REGRESSION,
+    DATASET_CCPP_REGRESSION,
+)
+
+
+class OpflowQnnRegressorBenchmarks(BaseRegressorBenchmark):
+    """Opflow QNN Regressor benchmarks."""
+
+    version = 1
+    timeout = 1200.0
+    params = [
+        [DATASET_SYNTHETIC_REGRESSION, DATASET_CCPP_REGRESSION],
+        ["qasm_simulator", "statevector_simulator"],
+    ]
+    param_names = ["dataset", "backend name"]
+
+    def __init__(self):
+        super().__init__()
+        self.model = None
+        self.train_features = None
+        self.train_labels = None
+        self.test_features = None
+        self.test_labels = None
+
+    def setup_cache(self):
+        """Cache Opflow fitted model"""
+        for dataset, backend in product(*self.params):
+            train_features = self.datasets[dataset]["train_features"]
+            train_labels = self.datasets[dataset]["train_labels"]
+
+            if dataset == DATASET_SYNTHETIC_REGRESSION:
+                model = self._construct_qnn_synthetic(
+                    quantum_instance_name=backend, optimizer=COBYLA()
+                )
+            else:
+                # we have only two dataset for now, so this clause is for "dataset_ccpp"
+                model = self._construct_qnn_ccpp(
+                    quantum_instance_name=backend, optimizer=COBYLA(maxiter=100)
+                )
```

Comment:
I would introduce an `elif` statement that checks for ccpp and throw an error in `else`.

Reply:
Please check other places for same issue.

Reply:
Done in the regression benchmarks. Will change in the classification benchmarks as well.